### PR TITLE
Add initial non-interactive framework

### PR DESCRIPTION
**Pull request by jaweesner**
Elements in general locations and populated with test data

### CODE REVIEW

**Comment by jaweesner** [comment]
[PR Jane's initial front End (up for grabs)](https://trello.com/c/P3qUGGYt/39-pr-janes-initial-front-end-up-for-grabs)

**Comment by jaweesner** [comment]
No one volunteered so @CareyL96, you're it :)

File: webpack.config.js [hunk 1]
```diff
@@ -11,15 +11,16 @@ module.exports = {
...
-          loader: 'babel-loader',
-        },
-      },
-    ],
-  },
+        test : /\.jsx?/,
+        exclude : /node_modules/,
+        loader : 'babel-loader',      
+        query: {
+          presets: ['react', 'es2015']
```
Comment (careyl96) [inline webpack.config.js]: I believe that the babel-preset-env is the updated version of babel-preset-es2015

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <meta charset="utf-8" />
+  <title>Tools</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+  <link rel="stylesheet" href="/style.css">
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+
+</head>
+<body>
+  <div id = "app" class= "container" ></div>
```
Comment (careyl96) [inline public/index.html]: no spaces: id = "app" should equal id="app"

File: client/components/ProductImages.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import {Grid, Row, Col,Image} from 'react-bootstrap'
```
Comment (careyl96) [inline client/components/ProductImages.jsx]: space after Col,

File: client/components/ProductImages.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      {props.images.map(image => 
+        (
+          <Col md={3} className = "no-gutter">
+            <Image src={image} responsive /> 
+          </Col>  
+        )
+      )}
+    </Row>
+  </Grid>
+)
```
Comment (careyl96) [inline client/components/ProductImages.jsx]: newline at end of file (airbnb styleguide + github points it out as well)

File: client/components/List.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import {ListGroup, ListGroupItem} from 'react-bootstrap'
+export const List = (props) => (
+  <ListGroup>
+    {props.tools.map((tool) => <SingleTool tool= {tool} key={tool.id}/>)}
+  </ListGroup>
+
```
Comment (careyl96) [inline client/components/List.jsx]: extra space

File: client/components/Detail.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    <Row className="show-grid">
+      <Col sm={6} md={6}>
+        <ProductImages images = {props.tool.imageurls}/>
+      </Col>
+      <Col sm={6} md={6}>
+        <ProductDetail tool = {props.tool}/>
+      </Col>
+    </Row>
+  </Grid>
+
```
Comment (careyl96) [inline client/components/Detail.jsx]: extra space

File: client/components/List.jsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import {ListGroup, ListGroupItem} from 'react-bootstrap'
+export const List = (props) => (
+  <ListGroup>
+    {props.tools.map((tool) => <SingleTool tool= {tool} key={tool.id}/>)}
```
Comment (careyl96) [inline client/components/List.jsx]: Two things: 
use object destructuring for your props; List = (props) --> List = ({tools})
remove space after equals sign: tool= {tool} --> tool={tool}

File: client/components/ProductImages.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import {Grid, Row, Col,Image} from 'react-bootstrap'
+
+export const ProductImages = (props) =>
+
+(
+  <Grid fluid>
+    <Row className="show-grid">
+      <Col md={12} className= "no-gutter">
```
Comment (careyl96) [inline client/components/ProductImages.jsx]: rm space after className

File: client/components/ProductImages.jsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  <Grid fluid>
+    <Row className="show-grid">
+      <Col md={12} className= "no-gutter">
+        <Image src={props.images[0]} responsive />
+      </Col>
+    </Row>
+    <Row>
+      {props.images.map(image => 
+        (
+          <Col md={3} className = "no-gutter">
```
Comment (careyl96) [inline client/components/ProductImages.jsx]: rm space after className and before "no-gutter"

File: client/components/ProductDetail.jsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        <h3>{props.tool.name}</h3> by <h4>{props.tool.manufacturer}</h4>
+      </Col>
+    </Row>
+    <Row>
+      <Col md={12}>
+        <p> {props.tool.description} </p>
+      </Col>
+    </Row>
+    <Row>
+      <Col md={2} mdPush= {4}>
```
Comment (careyl96) [inline client/components/ProductDetail.jsx]: rm space after =

File: client/components/ProductDetail.jsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        ${props.tool.price}
+      </Col>
+      <Col md={6} mdPush= {4}>
```
Comment (careyl96) [inline client/components/ProductDetail.jsx]: rm space after =

File: client/components/ProductDetail.jsx [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import {Grid, Row, Col, Button} from 'react-bootstrap'
+export const ProductDetail = (props) =>
```
Comment (careyl96) [inline client/components/ProductDetail.jsx]: use object destructuring: ProductDetail = (props) --> ProductDetail = ({tool})

File: client/components/List.jsx [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import {ListGroup, ListGroupItem} from 'react-bootstrap'
+export const List = (props) => (
+  <ListGroup>
+    {props.tools.map((tool) => <SingleTool tool= {tool} key={tool.id}/>)}
+  </ListGroup>
+
+)
+
+const SingleTool = (props) => (
```
Comment (careyl96) [inline client/components/List.jsx:10]: use object destructuring

File: client/components/Detail.jsx [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+import {ProductImages} from './ProductImages.jsx'
+
+export const Detail = (props) => (
+  <Grid fluid>
+    <Row className="show-grid">
+      <Col sm={6} md={6}>
+        <ProductImages images = {props.tool.imageurls}/>
+      </Col>
+      <Col sm={6} md={6}>
+        <ProductDetail tool = {props.tool}/>
```
Comment (careyl96) [inline client/components/Detail.jsx]: rm spaces between tool and props.tool

File: client/components/Detail.jsx [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import {Grid, Row, Col} from 'react-bootstrap'
+import {ProductDetail} from './ProductDetail.jsx'
+import {ProductImages} from './ProductImages.jsx'
+
+export const Detail = (props) => (
+  <Grid fluid>
+    <Row className="show-grid">
+      <Col sm={6} md={6}>
+        <ProductImages images = {props.tool.imageurls}/>
```
Comment (careyl96) [inline client/components/Detail.jsx]: remove spaces: images={props.tool.imageurls}

File: client/components/App.jsx [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+          <Row className="show-grid">
+          {/* TODO hide this and show drop down when v small */}
+          <Col md={4}>
+            <List tools = {this.state.tools} />
+          </Col>
+          <Col xs={12} md={8}>
+            <Detail tool = {this.state.currentTool} />
+          </Col>
+          </Row>
+
```
Comment (careyl96) [inline client/components/App.jsx]: extra space

File: client/components/App.jsx [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  
+
+  render(props){
+    return (
+        <Grid>
+          <Row>
+            <Col md={8}>
+            <h4> tried-and-true </h4>
+             <h2> Kitchen Tools </h2>
```
Comment (careyl96) [inline client/components/App.jsx]: indent h2 and h4

**Review by careyl96 — CHANGES_REQUESTED** [review]
Your code is very clean and readable; I'm able to easily distinguish what each component is responsible for. Other than some syntax issues, the only changes that I would recommend making are using object destructuring for your props and potentially renaming your list component to ToolList.

File: webpack.config.js (hunk 1 above)
Comment (jaweesner) [reply webpack.config.js]: Adding this in was an act of desperation when my webpack wasn't working. I have the presets it in my package json so I removed both lines.

**Comment by jaweesner** [comment]
@CareyL96  Fixed linting!

**Review by careyl96 — APPROVED** [review]
I see some potential object destructuring opportunities in your List component but everything else looks good!

**Pull request merged**